src/main/groovy/actions/TransferActions.groovy

```diff
@@ -82,15 +84,16 @@ class TransferActions {
         updateLabware(destinationLabware)
     }
 
-    private static getMaterialsFromLabware(materialUuids) {
+    private static getMaterialsByUuid(materialUuids) {
         MaterialActions.getMaterials(materialUuids)
     }
 
-    private static createNewChildMaterial(materialName, type, sourceMaterial, copyMetadata) {
+    private static createNewChildMaterial(materialName, type, sourceMaterial,
+        copyMetadata, newMetadataToAdd) {
         new Material(
             name: materialName,
             materialType: type,
-            metadata: sourceMaterial.metadata.findAll { it.key in copyMetadata },
+            metadata: (sourceMaterial.metadata.findAll { it.key in copyMetadata } << newMetadataToAdd).flatten(),
```

Comment:
You can just use + to concatenate the lists of metadata rather than appending and then flattening.